hack/update-crd.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#set -o errexit
+set -o nounset
+set -o pipefail
+
+## find or download controller-gen
+CONTROLLER_GEN=$(which controller-gen)
+
+if [ "$CONTROLLER_GEN" = "" ]
+then
+  TMP_DIR=$(mktemp -d);
+  cd $TMP_DIR;
+  go mod init tmp;
```

Comment:
what is this step supposed to do?

Reply:
this step allows "go get" to work under the temp module folder.